shared/zodSchemas.ts

```diff
@@ -0,0 +1,52 @@
+import { z } from 'zod';
+import { emailRegex, passwordRegex } from './regexPatterns';
+import { Prisma } from '@prisma/client';
+
+export const giftSchema = z.object({
+  gift: z.string().min(1),
+  receiver: z.string().min(1),
+  createdAt: z.date(),
+  updatedAt: z.date(),
+  uuid: z.string(),
+});
+
+export const createGiftSchema = z.object({
+  gift: z.string().min(1),
+  receiver: z.string().min(1),
+});
+
+// CHECK THIS
+// onko alla oleva rivi tarpeellinen?
+export const updateGiftSchema = z.object({
+  gift: z.string().min(1),
+  receiver: z.string().min(1),
+});
+
+export const userSchema = z.object({
+  createdAt: z.date(),
+  updatedAt: z.date(),
+  uuid: z.string(),
+  email: z.string().min(1).max(128).regex(emailRegex),
+  firstName: z.string().min(1).max(128),
+  lastName: z.string().min(1).max(128),
+});
```

Comment:
Skeemoja kannattaa extendata niin ei tarvitse määrittää samoja kenttiä moneen kertaan.
```suggestion
export const userSchema = z.object({
  email: z.string().min(1).max(128).regex(emailRegex),
  firstName: z.string().min(1).max(128),
  lastName: z.string().min(1).max(128),
});

export const getUserSchema = userSchema.extend({
  createdAt: z.date(),
  updatedAt: z.date(),
  uuid: z.string(),
});

export const updateUserSchema = userSchema;

export const createUserSchema = userSchema.extend({
  password: z.string().min(1).max(128).regex(passwordRegex),
});
```

Reply:
Hyvä esimerkki! Kiitos! :+1: Yksi kysymys tuli tuohon koskien, mutta kommentoin kun pistän re-review-pyynnön